### PR TITLE
Edit DG NF requirements and UG add and remove

### PR DESCRIPTION
**Pull request by Vinci-Hu**
plus added @@author tags

### CODE REVIEW

**Review by SimJJ96 — APPROVED** [review]
LGTM

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -167,7 +167,47 @@ _Exceptions related to file storage:_
...
+#### Constraint: Typing-Preferred
+The product should be targeting users who can type fast and prefer typing over other means of input.
+
+**Reason**: to increase comparability of products, and to make feature evaluation easier for peer evaluators.
+
+
+### Portability and compatibility
+
+#### Constraint: Platform-Independent
+The software should work on the Windows, Linux, and OS-X platforms. Even if you are unable to manually test the app on all three platforms, deliberately avoid using OS-dependent libraries and OS-specific features.
```
Comment (kwokyto) [inline docs/DeveloperGuide.md]: Make the last sentence in third person

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -167,7 +167,47 @@ _Exceptions related to file storage:_
...
+The software should work on a computer that has version 11 of Java i.e., no other Java version installed.
+
+### Reliability, availability, maintainability
+ As it is a locally-runned command line application, there will be no down time for users.
+ It should be available anytime and anywhere.
+
+#### Constraint-No-Remote-Server
+The software should not depend on your own remote server.
+
+**Reason**: Anyone should be able to use/test your app any time, even after the semester is over.
```
Comment (kwokyto) [inline docs/DeveloperGuide.md]: Make this sentence in third person

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -167,7 +167,47 @@ _Exceptions related to file storage:_
...
+
+#### Constraint: Java-Version
+The software should work on a computer that has version 11 of Java i.e., no other Java version installed.
+
+### Reliability, availability, maintainability
+ As it is a locally-runned command line application, there will be no down time for users.
+ It should be available anytime and anywhere.
+
+#### Constraint-No-Remote-Server
+The software should not depend on your own remote server.
```
Comment (kwokyto) [inline docs/DeveloperGuide.md]: Make this sentence in third person

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -167,7 +167,47 @@ _Exceptions related to file storage:_
...
+### Security
+
+#### Constraint: Human-Editable-File
+The data should be stored locally and should be in a human editable text file.
+Thus the data is not protected by nature.
+
+**Reason**: To allow advanced users to manipulate the data by editing the data file.
+
+### Usability
+ It is very easy for a user to use the system. There are detailed user guide provided as well as a help command.
```
Comment (kwokyto) [inline docs/DeveloperGuide.md]: There is a*

**Review by kwokyto — APPROVED** [review]
Other than those commented, LGTM

**Pull request merged**